documentation/guides-code-generation.asciidoc

```diff
@@ -21,131 +21,119 @@ As we mention in the page link:guide-key-principles[key principles], one of our
-CREATE /test-app/src/app/core/auth/strategies/index.ts (33 bytes)
-CREATE /test-app/src/app/core/auth/strategies/jwt.strategy.spec.ts (713 bytes)
-CREATE /test-app/src/app/core/auth/strategies/jwt.strategy.ts (618 bytes)
-CREATE /test-app/src/app/core/user/user.module.ts (338 bytes)
-CREATE /test-app/src/app/core/user/model/index.ts (82 bytes)
-CREATE /test-app/src/app/core/user/model/dto/user-payload.dto.ts (188 bytes)
-CREATE /test-app/src/app/core/user/model/entities/user.entity.ts (542 bytes)
-CREATE /test-app/src/app/core/user/services/index.ts (33 bytes)
-CREATE /test-app/src/app/core/user/services/user.service.spec.ts (2127 bytes)
-CREATE /test-app/src/app/core/user/services/user.service.ts (1065 bytes)
-CREATE /test-app/test/auth/auth.service.mock.ts (948 bytes)
-CREATE /test-app/test/user/user.repository.mock.ts (1095 bytes)
-CREATE /test-app/src/app/test/test.module.ts (469 bytes)
-CREATE /test-app/src/app/test/model/entities/test.entity.ts (174 bytes)
-CREATE /test-app/src/app/test/model/index.ts (88 bytes)
-CREATE /test-app/src/app/test/controllers/test.crud.controller.ts (393 bytes)
-CREATE /test-app/src/app/test/services/test.crud.service.ts (394 bytes)
-CREATE /test-app/src/app/test/services/index.ts (69 bytes)
-CREATE /test-app/src/app/test/controllers/index.ts (75 bytes)
-CREATE /test-app/src/app/test/controllers/test.controller.spec.ts (496 bytes)
-CREATE /test-app/src/app/test/controllers/test.controller.ts (107 bytes)
-CREATE /test-app/src/app/test/services/test.service.spec.ts (464 bytes)
-CREATE /test-app/src/app/test/services/test.service.ts (92 bytes)
-CREATE /test-app/src/app/test/model/entities/another-test.entity.ts (181 bytes)
-----
-
-Then, install all dependencies, format the code and initialize git.
-
-=== Generate more components
-
-The `generate` command has also an interactive mode. To execute `generate` command in interactive mode, you only need to change to an existing application and then execute the command `devon4node generate -i`
-
-image::images/generate-interactive.jpg[]
-
-The functionality is the same as new command, but the difference is `generate` command do not generate the base application.
+// == Differences between tools
+
+// As you may have noticed, we have two tools to generate code, instead of having just one. What is the reason? The reason is that each tool starts from a different input. devon4node CLI receives user input, while cobigen receives code already exists and generates the rest accordingly. So, in order to increase the productivity, we can combine the use of both.
+
+== Nest CLI and Devon4node schematics
+ 
+We are going to use the Nest CLI to generate code of our application, you can know more about NodeJs CLI in the official link:https://docs.nestjs.com/cli/overview:[documentation].
+
+=== Install devon4node schematics
+
+First of all, you need to install Nest CLI 
+
+Execute the command `yarn global add @nestjs/cli`
```

Comment:
In addition to the `yarn` commands, could you also mention the npm alternatives for installing `@nestjs/cli` and `@devon4node/schematics`. Personally `yarn` did not work for me (my machine wouldn't recognize the `nest` command even after multiple attempts on multiple machines).